src/store/generator/slices/recording.utils.ts

```diff
@@ -1,11 +1,24 @@
-import { uniq } from 'lodash-es'
+import { uniq, intersection, difference } from 'lodash-es'
 
 import { ProxyData } from '@/types'
 
 export function extractUniqueHosts(requests: ProxyData[]) {
   return uniq(requests.map((request) => request.request.host).filter(Boolean))
 }
 
+export function isHostThirdParty(host: string) {
+  const hostPatterns = ['.google.com', '.googleapis.com', '.gstatic.com']
+  return hostPatterns.some((pattern) => host.includes(pattern))
+}
+
+export function reorderHosts(hosts: string[]) {
+  const matchedHosts = hosts.filter((host) => isHostThirdParty(host))
+  const hostsToGoLast = intersection(hosts, matchedHosts)
+  const hostsToGoFirst = difference(hosts, hostsToGoLast)
+
+  return [...hostsToGoFirst, ...hostsToGoLast]
```

Comment:
`.googleapis.com` might be the first item in the HAR file due to Chrome automatically making requests but here we ensure that they are moved to the end of the array, leaving the first one to be a URL that the user actually requested.